### PR TITLE
Support of .git folder and .gitignore #82

### PR DESCRIPTION
**Pull request by apoorv-2204**
## Test
Tested with ae website
### Commands
1=>
``` bash 
 node aeweb-cli/aeweb.js generate-address --seed abc --index 0
 node aeweb-cli/aeweb.js generate-address --seed def --index 0

 ```
2=>With flag 31 txns
https://testnet.archethic.net/explorer/transaction/0000D0B974CB6DA0716011A475234072783B84F3026742AC5E960E0997090A759DAE

``` bash 
 node aeweb-cli/aeweb.js  deploy --path ae_git_site --endpoint https://testnet.archethic.net/ --seed abc --include-git-ignored-files
 ```

3=>Without flag 27txns
https://testnet.archethic.net/explorer/transaction/000018C50C40693E9402470EAAC9E0986D7D0E2E2EA69E436DCEDF09F9D4E307B15B
``` bash 
 node aeweb-cli/aeweb.js  deploy --path ae_git_site --endpoint https://testnet.archethic.net/ --seed def 
```

#### ae_git_site

image

#### .gitignore

image

### CODE REVIEW

**Comment by samuelmanzanera** [comment]
I got this error while trying to deploy on latest version of archethic-node#develop

image

**Comment by samuelmanzanera** [comment]
The .gitignore or the node_modules is still uploaded

image

**Comment by apoorv-2204** [comment]
> I got this error while trying to deploy on latest version of archethic-node#develop
> 
> image

testnet not updated

**Comment by apoorv-2204** [comment]
> The .gitignore or the node_modules is still uploaded
> 
> image

if you put a flag,then it will still be there.

**Comment by samuelmanzanera** [comment]
> > I got this error while trying to deploy on latest version of archethic-node#develop
> > image
> 
> testnet not updated

It's not on testnet, it's on the latest version of archethic-node  (develop branch)

**Comment by samuelmanzanera** [comment]
> > The .gitignore or the node_modules is still uploaded
> > image
> 
> if you put the flag,it will be there.

I didn't put the flag

**Comment by apoorv-2204** [comment]
> > > The .gitignore or the node_modules is still uploaded
> > > image
> > 
> > 
> > if you put the flag,it will be there.
> 
> I didn't put the flag

please share , .gitignore, and content to upload.

File: commands/cli.js [hunk 1]
```diff
@@ -66,17 +51,58 @@ export async function estimateTxsFees(archethic, transactions) {
...
+
+  const isGitIgnored = ignore().add(filters)
+  doHandleDirectory(entry, files, isGitIgnored)
+}
+
+function doHandleDirectory(entry, files, isGitIgnored) {
+  // reduce search space by omitting folders at once
+  if (fs.statSync(entry).isDirectory() && !isGitIgnored.ignores(entry)) {
+    fs.readdirSync(entry).forEach((child) => {
+      doHandleDirectory(entry + PathLib.sep + child, files, isGitIgnored)
```
Comment (bchamagne) [inline commands/cli.js:94]: you are using files (`entry + PathLib.sep + child`) on a function called `doHandleDirectory`, it's very confusing. I suggest renaming the `entry` to `folderPath` for explicitness.

Then I suggest not to call that recursion with files.

```javascript
function doHandleDirectory(folderPath, files, isGitIgnored) {
  // reduce search space by omitting folders at once
  if (!isGitIgnored.ignores(folderPath)) {
    fs.readdirSync(folderPath).forEach((fileOrFolderPath) => {
      if (fs.statSync(fileOrFolderPath).isDirectory()) {
        doHandleDirectory(fileOrFolderPath, files, isGitIgnored)
      } else {
        doHandleFile(filePath, files, isGitIgnored); 
      }
    })
  } 
}
```

**Comment by apoorv-2204** [comment]
> > > The .gitignore or the node_modules is still uploaded
> 
> > if you put the flag,it will be there.
> 
> I didn't put the flag

Hello. This has been resolved in.
https://github.com/archethic-foundation/aeweb-cli/pull/95

**Pull request merged**